File: Python/fizzbuzz4.py
from functools import partial

LIMIT = 100

print = partial(print, end='')

for i in range(1, LIMIT + 1):
    match i % 3, i % 5:
        case 0, 0: print('FizzBuzz')
        case 0, _: print('Fizz')
        case _, 0: print('Buzz')
        case _: print(i)
    if i < LIMIT:
        print(', ')

print('\n')

File: Go/fizzbuzz4.go
package main

import (
	"fmt"
)

const LIMIT = 100

func main() {
	a := [4]string{"", "Fizz", "Buzz", "FizzBuzz"}
	b := [15]int{3: 1, 6: 1, 9: 1, 12: 1, 5: 2, 10: 2, 0: 3}

	for i := 1; i <= LIMIT; i++ {
		if s := a[b[i%15]]; len(s) > 0 {
			fmt.Print(s)
		} else {
			fmt.Print(i)
		}
		if i < LIMIT {
			fmt.Print(", ")
		}
	}
	fmt.Println()
}
